src/engine/world/room.h

```diff
@@ -47,6 +48,7 @@ namespace loader::file
 {
 struct Portal;
 struct Room;
+struct RoomVertex;
```

Comment:
Forward declaration is not enough, `std::vector` needs a complete type. Should become obsolete with the other changes I suggested.